iFixit/react-commerce
# Add zoom to product gallery

closes #726 

This PR adds the side zoom functionality on product page.

<img width="1277" alt="Screenshot 2022-09-29 at 15 48 56" src="https://user-images.githubusercontent.com/7805759/193053234-0bfb60b4-c486-477f-ab66-e8bab6d02c66.png">

### Desktop/large screens 
Just hover with the mouse over any image in the gallery and a zoom side panel will appear.

### Mobile/small screens
We reverted #196 since now it is possible to handle pinch-zooming on modals - while still avoiding background scroll - exploiting the `allowPinchZoom ` flag of the [Chakra Modal component](https://chakra-ui.com/docs/components/modal/props).
So on mobile it is now possible to zoom on the gallery image by simply pinch zooming on the page.
Consider this as a base implementation for mobile. If then we want to add a more complex behaviour for mobile (click to fullscreen) we can discuss and integrate in a separate PR

### QA

1. Open Vercel preview
2. Verify that the zoom behaviour is working correctly on desktop
3. Verify that on mobile it is possible to pinch zoom
